awx/api/serializers.py

```diff
@@ -2515,17 +2554,26 @@ def validate_extra_vars(self, value):
 
     def get_summary_fields(self, obj):
         summary_fields = super(JobTemplateSerializer, self).get_summary_fields(obj)
-        if 'pk' in self.context['view'].kwargs and self.version > 1:  # TODO: remove version check in 3.3
+        if self.is_detail_view and self.version > 1:  # TODO: remove version check in 3.3
+            all_creds = []
             extra_creds = []
-            for cred in obj.extra_credentials.all():
-                extra_creds.append({
+            for cred in obj.credentials.all():
+                summarized_cred = {
                     'id': cred.pk,
                     'name': cred.name,
                     'description': cred.description,
                     'kind': cred.kind,
                     'credential_type_id': cred.credential_type_id
-                })
+                }
+                all_creds.append(summarized_cred)
+                if cred.credential_type.kind in ('cloud', 'net'):
+                    extra_creds.append(summarized_cred)
+                elif cred.credential_type.kind == 'ssh':
+                    summary_fields['credential'] = summarized_cred
+                elif cred.credential_type.kind == 'vault':
+                    summary_fields['vault_credential'] = summarized_cred
             summary_fields['extra_credentials'] = extra_creds
+            summary_fields['credentials'] = all_creds
```

Comment:
`summary_fields` contain `credentials` *and* the deprecated keys (`credential`, `vault_credential`, `extra_credentials`)